File: java/BinarySearch.java
class BinarySearch{  

    public static int binary_search(int arr[],int find_element){
        
        int array_size=arr.length;
        int high=array_size;
        int low=0;
        int element_index=-1;

        while(low<high){

              int midIndex=(high+low)/2;

              if(arr[midIndex]==find_element){
                  element_index=midIndex;
                  break;
              }else if(find_element<arr[midIndex]){
                    high=midIndex;
              }else if(find_element>arr[midIndex]){
                    low=midIndex;
              }


        }
        
        return element_index;
    }

    public static void main(String args[]){  
     int array[]={1,2,3,4,5,6,7};
     int search_element=5;  
     int result=binary_search(array,search_element);

        if(result!=-1){
            System.out.print("Element "+search_element+" found on index "+result);
        }else{
            System.out.print("Element "+search_element+" not found");
        }
    }  
}  

File: js/binary_search.js
const arr=[2,44,55,66,77,88,99,100,200];
var element=44;
result=search_element(arr,element);
if(result!=-1)
console.log("Element "+element+" found on index "+result);
else
console.log("Element "+element+" not found ");

function search_element(arr,element){
   var result=-1; 
   var high=arr.length;
   var low=0;

   while(low<high){
      var mid =parseInt((high+low)/2);
      if(element==arr[mid]){
          result=mid;
          break;
      }else if(element<arr[mid]){
          high=mid;
      }else if(element>arr[mid]){
          low=mid;
      }
   }

   return result;
}